src/common/options/crimson.yaml.in

```diff
@@ -258,3 +258,8 @@ options:
         internal checksum feature without using sever CPU then enable if available,
         set to true to disable unconditionally.
   default: true
+- name: seastore_enable_shard_change
+  type: bool
+  level: advanced
+  desc: enable osd shards changes upon restart.
+  default: true
```

Comment:
Ultimately, I think we should expect this to be set off. Once a cluster needs to change, one can change this option to true. Alternatively, if we decide to keep this true by default - we might need to reconsider the purpose of introducing this option in the first place. @liu-chunmei, @athanatos - What do you think?

Moreover, We will need to make sure that no unexpected behavior is there if this options is enabled, shard number changes and then this is set back off.

Reply:
here double confirm:  change option name from seastore_enable_shard_change to seastore_require_partition_count_match_reactor_count,  
and default is true equal to seastore_enable_shard_change's false, right?

Reply:
Yes